### PR TITLE
Update vertex quickstart to use correct API

### PR DESCRIPTION
**Pull request by rlazo**
The API of the underlying vertex SDK has changed.

### CODE REVIEW

File: vertexai/app/build.gradle.kts [hunk 1]
```diff
@@ -56,16 +57,16 @@ dependencies {
     implementation("androidx.lifecycle:lifecycle-viewmodel-compose:2.8.6")
     implementation("androidx.activity:activity-compose:1.9.2")
     implementation("androidx.navigation:navigation-compose:2.8.2")
+    implementation("org.jetbrains.kotlinx:kotlinx-serialization-json:1.7.3")
```
[the remark this answers is not in the document]
Comment (rlazo) [reply vertexai/app/build.gradle.kts:60]: It's required for function calling. The data returned from the model is represented by the Json serialization objects

File: vertexai/app/build.gradle.kts [hunk 2]
```diff
@@ -56,16 +57,16 @@ dependencies {
...
+
     implementation("androidx.compose.ui:ui")
     implementation("androidx.compose.ui:ui-graphics")
     implementation("androidx.compose.ui:ui-tooling-preview")
     implementation("androidx.compose.material3:material3")
     implementation("io.coil-kt:coil-compose:2.7.0")
-    
-    implementation("com.google.firebase:firebase-analytics:22.1.2")
-    implementation("com.google.firebase:firebase-vertexai:16.0.0-beta06")
+    implementation("com.google.firebase:firebase-vertexai:16.0.0")
```
[the remark this answers is not in the document]
Comment (rlazo) [reply vertexai/app/build.gradle.kts:69]: Modified to use the BOM.

Using appcheck sounds like an interesting idea, but can be explored in a follow up PR.

File: vertexai/app/src/main/kotlin/com/google/firebase/quickstart/vertexai/GenerativeAiViewModelFactory.kt [hunk 3]
```diff
@@ -66,31 +65,29 @@ val GenerativeViewModelFactory = object : ViewModelProvider.Factory {
...
                         generationConfig = config
                     )
                     ChatViewModel(generativeModel)
                 }
 
                 isAssignableFrom(FunctionsChatViewModel::class.java) -> {
                     // Declare the functions you want to make available to the model
+                    val functionDeclaration = FunctionDeclaration(
+                        "upperCase", "Returns the upper case version of the input string", mapOf(
+                            "input" to Schema.string( "Text to transform")))
```
[the remark this answers is not in the document]
Comment (rlazo) [reply vertexai/app/src/main/kotlin/com/google/firebase/quickstart/vertexai/GenerativeAiViewModelFactory.kt:78]: Done

File: vertexai/app/src/main/kotlin/com/google/firebase/quickstart/vertexai/GenerativeAiViewModelFactory.kt [hunk 4]
```diff
@@ -66,31 +65,29 @@ val GenerativeViewModelFactory = object : ViewModelProvider.Factory {
...
+                                functionDeclaration
                             )
                         )
                     )
 
+
                     // Initialize a GenerativeModel with the `gemini-pro` AI model for function calling chat
                     val generativeModel = Firebase.vertexAI.generativeModel(
-                        modelName = "gemini-1.5-pro-preview-0514",
+                        modelName = "gemini-1.5-pro",
```
[the remark this answers is not in the document]
Comment (rlazo) [reply vertexai/app/src/main/kotlin/com/google/firebase/quickstart/vertexai/GenerativeAiViewModelFactory.kt:90]: done

**Review by thatfiredev — APPROVED** [review]
LGTM

**Pull request merged**